### DISCUSSION TITLE
Delay leaving call when queue is empty?

### DISCUSSION THREAD

**Discussion by rundagger** [TannerGabriel/discord-bot#132] [Q&A]
Hello, I attempted to follow your instructions on your blog but am unsure where to put the following code to have the bot delay leaving the call when the queue is empty:
```
setTimeout(function() {
       serverQueue.voiceChannel.leave()
}, 10000);
```

I tried to put under the index.js 
```
player.on('queueEnd', queue => {
  queue.metadata.send('✅ | Queue finished!');
  setTimeout(function() {
    serverQueue.voiceChannel.leave()
}, 10000);
});
```
But then I get the error that serverQueue is not defined. Am I putting this in the right place?

**Comment by TannerGabriel** [comment]
The bot on Github currently uses a newer `discordjs` version than the one in the article on my blog. Therefore the code mentioned above will not work here. Since we are not manually leaving the channel like with the blog version of the bot, but instead use the [discord-player](https://github.com/Androz2091/discord-player) library, which has the leaving functionality implemented internally we will have to change something there. 

I have looked for configurable parameters that would make a delay possible but currently could not find any. I will take a closer look at a later point. If the functionality is not implemented yet, I will probably implement it myself then.
